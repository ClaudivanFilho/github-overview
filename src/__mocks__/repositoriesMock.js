export default [
  {
    id: '1',
    description: 'description',
    url: 'http://github.com/vtex-apps/render-runtime',
    nameWithOwner: 'vtex-apps/render-runtime',
    issues: {
      totalCount: 0,
      nodes: [],
    },
    refs: {
      totalCount: 0,
      nodes: [],
    },
    pullRequests: {
      totalCount: 0,
      nodes: [],
    },
    collaborators: {
      totalCount: 0,
      nodes: [],
    },
  },
]

export const RepositoryMock = {
  'id': 'MDEwOlJlcG9zaXRvcnkxMjkxNDk3Nzk=',
  'description': 'VTEX Minicart app ',
  'createdAt': '2018-04-11T20:23:43Z',
  'url': 'https://github.com/vtex-apps/minicart',
  'nameWithOwner': 'vtex-apps/minicart',
  'collaborators': {
    'totalCount': 55,
    'nodes': [
      {
        'url': 'https://github.com/rsimoens',
        'avatarUrl': 'https://avatars3.githubusercontent.com/u/10401309?v=4',
        'name': 'Raphael Simoens',
      },
      {
        'url': 'https://github.com/vitoriaHeliane',
        'avatarUrl': 'https://avatars3.githubusercontent.com/u/12852518?v=4',
        'name': 'Vitória Heliane',
      },
      {
        'url': 'https://github.com/lcfpadilha',
        'avatarUrl': 'https://avatars1.githubusercontent.com/u/13875462?v=4',
        'name': 'Leonardo Padilha',
      },
      {
        'url': 'https://github.com/mrkmat',
        'avatarUrl': 'https://avatars2.githubusercontent.com/u/14097431?v=4',
        'name': 'Mark Matern',
      },
      {
        'url': 'https://github.com/julopesferr',
        'avatarUrl': 'https://avatars1.githubusercontent.com/u/15784157?v=4',
        'name': 'Juliane Lopes Ferreira',
      },
      {
        'url': 'https://github.com/estacioneto',
        'avatarUrl': 'https://avatars2.githubusercontent.com/u/15948386?v=4',
        'name': 'Estácio Pereira',
      },
      {
        'url': 'https://github.com/luciannojunior',
        'avatarUrl': 'https://avatars3.githubusercontent.com/u/18706156?v=4',
        'name': 'Luciano de Oliveira Júnior',
      },
      {
        'url': 'https://github.com/gabrielgalc',
        'avatarUrl': 'https://avatars2.githubusercontent.com/u/21203990?v=4',
        'name': null,
      },
      {
        'url': 'https://github.com/lowbeer',
        'avatarUrl': 'https://avatars0.githubusercontent.com/u/23402009?v=4',
        'name': null,
      },
      {
        'url': 'https://github.com/luizerasmo',
        'avatarUrl': 'https://avatars0.githubusercontent.com/u/30424413?v=4',
        'name': null,
      },
      {
        'url': 'https://github.com/iduleba',
        'avatarUrl': 'https://avatars0.githubusercontent.com/u/31371069?v=4',
        'name': 'Ian Duleba',
      },
    ],
  },
  'refs': {
    'totalCount': 4,
    'nodes': [
      {
        'name': 'bugfix/grapqlerror',
        'repository': {
          'nameWithOwner': 'vtex-apps/minicart',
        },
        'target': {
          'history': {
            'edges': [
              {
                'node': {
                  'committedDate': '2018-05-12T18:58:01Z',
                  'author': {
                    'name': 'Felipe Sales',
                    'avatarUrl': 'https://avatars1.githubusercontent.com/u/3926634?v=4',
                  },
                },
              },
            ],
          },
        },
      },
      {
        'name': 'fix/duplicate-product',
        'repository': {
          'nameWithOwner': 'vtex-apps/minicart',
        },
        'target': {
          'history': {
            'edges': [
              {
                'node': {
                  'committedDate': '2018-07-03T14:20:39Z',
                  'author': {
                    'name': 'Lucas Cordeiro',
                    'avatarUrl': 'https://avatars3.githubusercontent.com/u/10223856?v=4',
                  },
                },
              },
            ],
          },
        },
      },
      {
        'name': 'master',
        'repository': {
          'nameWithOwner': 'vtex-apps/minicart',
        },
        'target': {
          'history': {
            'edges': [
              {
                'node': {
                  'committedDate': '2018-08-24T15:01:26Z',
                  'author': {
                    'name': 'Bruno Dias',
                    'avatarUrl': 'https://avatars0.githubusercontent.com/u/8000984?v=4',
                  },
                },
              },
            ],
          },
        },
      },
      {
        'name': 'update/quantity-selector',
        'repository': {
          'nameWithOwner': 'vtex-apps/minicart',
        },
        'target': {
          'history': {
            'edges': [
              {
                'node': {
                  'committedDate': '2018-08-29T12:46:12Z',
                  'author': {
                    'name': 'Vitória Heliane',
                    'avatarUrl': 'https://avatars3.githubusercontent.com/u/12852518?v=4',
                  },
                },
              },
            ],
          },
        },
      },
    ],
  },
  'stargazers': {
    'totalCount': 2,
  },
  'issues': {
    'totalCount': 0,
    'nodes': [],
  },
  'pullRequests': {
    'totalCount': 1,
    'nodes': [
      {
        'title': 'Update the Quantity Selector UI and Fix change quantity',
        'url': 'https://github.com/vtex-apps/minicart/pull/37',
        'mergeable': 'MERGEABLE',
        'assignees': {
          'nodes': [
            {
              'avatarUrl': 'https://avatars3.githubusercontent.com/u/12852518?v=4',
              'name': 'Vitória Heliane',
            },
          ],
        },
        'reviews': {
          'nodes': [
            {
              'author': {
                'login': 'ClaudivanFilho',
                'avatarUrl': 'https://avatars2.githubusercontent.com/u/4960686?v=4',
              },
              'state': 'CHANGES_REQUESTED',
            },
            {
              'author': {
                'login': 'salesfelipe',
                'avatarUrl': 'https://avatars1.githubusercontent.com/u/3926634?v=4',
              },
              'state': 'APPROVED',
            },
            {
              'author': {
                'login': 'ClaudivanFilho',
                'avatarUrl': 'https://avatars2.githubusercontent.com/u/4960686?v=4',
              },
              'state': 'CHANGES_REQUESTED',
            },
            {
              'author': {
                'login': 'ClaudivanFilho',
                'avatarUrl': 'https://avatars2.githubusercontent.com/u/4960686?v=4',
              },
              'state': 'APPROVED',
            },
          ],
        },
      },
    ],
  },
}
