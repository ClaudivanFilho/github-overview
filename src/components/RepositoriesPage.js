import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Repository from './Repository'
import OrderBy from './OrderBy'
import Search from './Search'

export default class RepositoriesPage extends Component {
  state = {
    repositories: null,
  }

  static propTypes = {
    repositories: PropTypes.arrayOf(PropTypes.object),
  }

  setRepositories = (repositories) => {
    this.setState({ repositories })
  }

  render() {
    const reps = this.state.repositories || this.props.repositories
    return (
      <div className="flex flex-column w-100 pa3 sans-serif">
        <div className="w-100 pb2">
          <div className="fl">
            <Search
              setRepositories={this.setRepositories}
              repositories={this.props.repositories}
            />
          </div>
          <div className="fr">
            <OrderBy
              setRepositories={this.setRepositories}
              repositories={reps}
            />
          </div>
        </div>
        <div className="w-100">
          {reps.map(rep => (
            <Repository key={rep.id} repository={rep} />
          ))}
        </div>
      </div>
    )
  }
}
