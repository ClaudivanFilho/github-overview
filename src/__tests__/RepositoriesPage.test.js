import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import RepositoriesPage from '../components/RepositoriesPage'
import repositoriesMock from '../__mocks__/repositoriesMock'

describe('Test RepositoriesPage component', () => {
  let mountedComponent = null

  beforeAll(() => {
    mountedComponent = mount(
      <RepositoriesPage repositories={repositoriesMock} />
    )
  })

  it('Renders correctly', () => {
    expect(toJson(mountedComponent)).toMatchSnapshot()
  })

  it('Component should be defined', () => {
    expect(mountedComponent).toBeDefined()
  })

  it('Must have one repository', () => {
    expect(mountedComponent.find('Repository').exists()).toBe(true)
    expect(mountedComponent.find('Repository')).toHaveLength(1)
  })

  it('Must have a OrderBy Component', () => {
    expect(mountedComponent.find('OrderBy').exists()).toBe(true)
  })

  it('Must have a Search Component', () => {
    expect(mountedComponent.find('Search').exists()).toBe(true)
  })

  it('Must have an AddRepository Component', () => {
    expect(mountedComponent.find('AddRepository').exists()).toBe(true)
  })
})
