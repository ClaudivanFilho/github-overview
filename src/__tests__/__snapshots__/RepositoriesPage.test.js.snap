// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RepositoriesPage component Renders correctly 1`] = `
<RepositoriesPageWithContext
  repositories={
    Array [
      Object {
        "collaborators": Object {
          "nodes": Array [],
          "totalCount": 0,
        },
        "description": "description",
        "id": "1",
        "issues": Object {
          "nodes": Array [],
          "totalCount": 0,
        },
        "nameWithOwner": "vtex-apps/render-runtime",
        "pullRequests": Object {
          "nodes": Array [],
          "totalCount": 0,
        },
        "refs": Object {
          "nodes": Array [],
          "totalCount": 0,
        },
        "url": "http://github.com/vtex-apps/render-runtime",
      },
    ]
  }
>
  <RepositoriesPage
    repositories={
      Array [
        Object {
          "collaborators": Object {
            "nodes": Array [],
            "totalCount": 0,
          },
          "description": "description",
          "id": "1",
          "issues": Object {
            "nodes": Array [],
            "totalCount": 0,
          },
          "nameWithOwner": "vtex-apps/render-runtime",
          "pullRequests": Object {
            "nodes": Array [],
            "totalCount": 0,
          },
          "refs": Object {
            "nodes": Array [],
            "totalCount": 0,
          },
          "url": "http://github.com/vtex-apps/render-runtime",
        },
      ]
    }
  >
    <div
      className="flex flex-column w-100 pa3 sans-serif"
    >
      <div
        className="w-100 flex flex-column flex-row-m pb2"
      >
        <div
          className="w-100 w-33-m flex"
        >
          <ComponentWithContext>
            <AddRepository
              openToast={[Function]}
              refetchRepositories={[Function]}
            >
              <div
                className="flex items-center w-100 w-auto-m ma2 ma0-m"
              >
                <input
                  className="input-reset db w-100 w-auto-m pa0 pl2 bg-white br2 ba b--silver"
                  id="repository-name"
                  onChange={[Function]}
                  placeholder="Owner/Name"
                  title="Add Repository"
                  value=""
                />
                <img
                  alt="add-icon"
                  className="flex items-center br7 ml2 pointer dim"
                  onClick={[Function]}
                  src="add.svg"
                  width="25"
                />
              </div>
            </AddRepository>
          </ComponentWithContext>
        </div>
        <div
          className="w-100 w-33-m flex justify-center"
        >
          <ComponentWithContext>
            <Search
              originalRepositories={Array []}
              updateRepositories={[Function]}
            >
              <div
                className="flex items-center w-100 w-auto-m ma2 ma0-m"
              >
                <input
                  className="input-reset ba b--silver db w-100 bg-white br2 pl8"
                  id="search"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="text"
                  value=""
                />
                <img
                  alt="s-icon"
                  className="pl3 absolute"
                  src="glass.svg"
                  width="20"
                />
              </div>
            </Search>
          </ComponentWithContext>
        </div>
        <div
          className="w-100 w-33-m ml-auto flex justify-end"
        >
          <ComponentWithContext>
            <OrderBy
              originalRepositories={Array []}
              repositories={Array []}
              updateRepositories={[Function]}
            >
              <div
                className="flex items-center justify-end w-100 w-auto-m ma2 ma0-m"
              >
                <select
                  className="w-100 w4-m input-reset pl2 bg-white mid-gray b--silver ba br2 pointer f5"
                  onChange={[Function]}
                >
                  <option
                    key="NAME_ASC"
                    value="NAME_ASC"
                  >
                    Name ASC
                  </option>
                  <option
                    key="NAME_DESC"
                    value="NAME_DESC"
                  >
                    Name DESC
                  </option>
                  <option
                    key="ISSUES_ASC"
                    value="ISSUES_ASC"
                  >
                    Issues ASC
                  </option>
                  <option
                    key="ISSUES_DESC"
                    value="ISSUES_DESC"
                  >
                    Issues DESC
                  </option>
                  <option
                    key="PRS_ASC"
                    value="PRS_ASC"
                  >
                    PR's ASC
                  </option>
                  <option
                    key="PRS_DESC"
                    value="PRS_DESC"
                  >
                    PR's DESC
                  </option>
                </select>
              </div>
            </OrderBy>
          </ComponentWithContext>
        </div>
      </div>
      <div
        className="w-100"
      >
        <RepositoryWithContext
          key="vtex-apps/render-runtime"
          repository={
            Object {
              "collaborators": Object {
                "nodes": Array [],
                "totalCount": 0,
              },
              "description": "description",
              "id": "1",
              "issues": Object {
                "nodes": Array [],
                "totalCount": 0,
              },
              "nameWithOwner": "vtex-apps/render-runtime",
              "pullRequests": Object {
                "nodes": Array [],
                "totalCount": 0,
              },
              "refs": Object {
                "nodes": Array [],
                "totalCount": 0,
              },
              "url": "http://github.com/vtex-apps/render-runtime",
            }
          }
        >
          <Repository
            openToast={[Function]}
            repository={
              Object {
                "collaborators": Object {
                  "nodes": Array [],
                  "totalCount": 0,
                },
                "description": "description",
                "id": "1",
                "issues": Object {
                  "nodes": Array [],
                  "totalCount": 0,
                },
                "nameWithOwner": "vtex-apps/render-runtime",
                "pullRequests": Object {
                  "nodes": Array [],
                  "totalCount": 0,
                },
                "refs": Object {
                  "nodes": Array [],
                  "totalCount": 0,
                },
                "url": "http://github.com/vtex-apps/render-runtime",
              }
            }
          >
            <div
              className="near-black ba bw1 b--black-20 br2 mb1"
            >
              <div
                className="bg-near-white pointer pa2 fw6 flex justify-between items-center"
                onClick={[Function]}
              >
                <div
                  className="flex w-90 w-30-m items-center"
                >
                  <img
                    alt="repo-icon"
                    className="ml2 mr4"
                    src="repo.svg"
                    width="25"
                  />
                  <a
                    className="truncate"
                    href="http://github.com/vtex-apps/render-runtime"
                    onClick={[Function]}
                    target="_blank"
                  >
                    vtex-apps/render-runtime
                  </a>
                </div>
                <label
                  className="w-20 f6 b dn db-m mr2"
                >
                  Issues (
                  0
                  )
                </label>
                <label
                  className="w-20 f6 b dn db-m mr2"
                >
                  Pull Requests (
                  0
                  )
                </label>
                <label
                  className="w-20 f6 b dn db-m mr2"
                >
                  Active Branches (
                  0
                  )
                </label>
                <ComponentWithContext
                  repository={
                    Object {
                      "collaborators": Object {
                        "nodes": Array [],
                        "totalCount": 0,
                      },
                      "description": "description",
                      "id": "1",
                      "issues": Object {
                        "nodes": Array [],
                        "totalCount": 0,
                      },
                      "nameWithOwner": "vtex-apps/render-runtime",
                      "pullRequests": Object {
                        "nodes": Array [],
                        "totalCount": 0,
                      },
                      "refs": Object {
                        "nodes": Array [],
                        "totalCount": 0,
                      },
                      "url": "http://github.com/vtex-apps/render-runtime",
                    }
                  }
                >
                  <RemoveRepository
                    openToast={[Function]}
                    refetchRepositories={[Function]}
                    repository={
                      Object {
                        "collaborators": Object {
                          "nodes": Array [],
                          "totalCount": 0,
                        },
                        "description": "description",
                        "id": "1",
                        "issues": Object {
                          "nodes": Array [],
                          "totalCount": 0,
                        },
                        "nameWithOwner": "vtex-apps/render-runtime",
                        "pullRequests": Object {
                          "nodes": Array [],
                          "totalCount": 0,
                        },
                        "refs": Object {
                          "nodes": Array [],
                          "totalCount": 0,
                        },
                        "url": "http://github.com/vtex-apps/render-runtime",
                      }
                    }
                  >
                    <img
                      alt="remove"
                      className="flex items-center br7 mr3 pointer dim"
                      onClick={[Function]}
                      src="remove.svg"
                      title="remove"
                      width="20"
                    />
                  </RemoveRepository>
                </ComponentWithContext>
              </div>
              <Collapse
                fixedHeight={-1}
                forceInitialAnimation={false}
                hasNestedCollapse={false}
                isOpened={false}
                onMeasure={[Function]}
                onRender={[Function]}
                onRest={[Function]}
                springConfig={
                  Object {
                    "damping": 20,
                    "stiffness": 100,
                  }
                }
                style={Object {}}
                theme={
                  Object {
                    "collapse": "ReactCollapse--collapse",
                    "content": "ReactCollapse--content",
                  }
                }
              >
                <Motion
                  defaultStyle={
                    Object {
                      "height": 0,
                    }
                  }
                  onRest={[Function]}
                  style={
                    Object {
                      "height": 0,
                    }
                  }
                >
                  <div
                    className="ReactCollapse--collapse"
                    style={
                      Object {
                        "height": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <div
                      className="ReactCollapse--content"
                    >
                      <div
                        className="flex flex-column flex-row-m"
                      >
                        <div
                          className="w-100 w-20-m pa2"
                        >
                          <div
                            className="flex flex-column"
                          >
                            <h5
                              className="mv2"
                            >
                              Collaborators
                            </h5>
                            <div />
                          </div>
                        </div>
                        <div
                          className="w-100 w-80-m flex flex-column pb3"
                        >
                          <div
                            className="w-100 flex flex-column pv2"
                          >
                            <h5
                              className="w-100 mv2"
                            >
                              Description:
                            </h5>
                            <p
                              className="f7"
                            >
                              description
                            </p>
                          </div>
                          <div
                            className="w-100 br mb2 o-10 ba b--mid-gray"
                          />
                          <div
                            className="w-100 flex flex-column flex-row-m"
                          >
                            <div
                              className="w-100 w-40-m flex flex-column"
                            >
                              <h5
                                className="w-100 mv2"
                              >
                                Active Branches (
                                0
                                )
                              </h5>
                              <div
                                className="overflow-y-auto f6"
                                style={
                                  Object {
                                    "maxHeight": "9rem",
                                  }
                                }
                              >
                                <BranchList
                                  branches={
                                    Object {
                                      "nodes": Array [],
                                      "totalCount": 0,
                                    }
                                  }
                                  dateColor=""
                                  seeMore={true}
                                />
                              </div>
                            </div>
                            <div
                              className="h-90 br mh2 o-10 ba b--mid-gray"
                            />
                            <div
                              className="w-100 w-60-m flex flex-column"
                            >
                              <h5
                                className="w-100 mv2"
                              >
                                PRS (
                                0
                                )
                              </h5>
                              <div
                                className="overflow-y-auto f6"
                                style={
                                  Object {
                                    "maxHeight": "9rem",
                                  }
                                }
                              >
                                <PRList
                                  nameWithOwner="vtex-apps/render-runtime"
                                  pullRequests={
                                    Object {
                                      "nodes": Array [],
                                      "totalCount": 0,
                                    }
                                  }
                                  seeMore={true}
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            className="w-100 br mv2 o-10 ba b--mid-gray"
                          />
                          <div
                            className="flex flex-column"
                          >
                            <h5
                              className="w-100 mv2"
                            >
                              Issues (
                              0
                              )
                            </h5>
                            <div
                              className="overflow-y-auto"
                              style={
                                Object {
                                  "maxHeight": "8rem",
                                }
                              }
                            >
                              <BranchList
                                issues={
                                  Object {
                                    "nodes": Array [],
                                    "totalCount": 0,
                                  }
                                }
                                nameWithOwner="vtex-apps/render-runtime"
                                seeMore={true}
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </Motion>
              </Collapse>
            </div>
          </Repository>
        </RepositoryWithContext>
      </div>
    </div>
  </RepositoriesPage>
</RepositoriesPageWithContext>
`;
